components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/auth/OIDCOauthManager.java

```diff
@@ -175,30 +174,99 @@ private JSONObject retrieveClientCredentials(String dcrEp, HttpClient client) th
         }
     }
 
+    /**
+     * This method will be used to validate access token.
+     *
+     * @param token This will be the access token.
+     * @return the boolean status of the validity of the access token.
+     */
     public boolean validateToken(String token) throws OIDCProviderException {
         try {
-            Unirest.setHttpClient(Utils.getAllSSLClient());
-            com.mashape.unirest.http.HttpResponse<String> stringResponse
-                    = Unirest.post(CelleryConfig.getInstance().getIdpURL() + Constants.INTROSPECT_ENDPOINT)
-                    .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_FORM_URLENCODED)
-                    .basicAuth(CelleryConfig.getInstance().getIdpAdminUsername()
-                            , CelleryConfig.getInstance().getIdpAdminPassword()).body("token=" + token).asString();
-
-            JSONObject jsonResponse = new JSONObject(stringResponse.getBody());
-            if (stringResponse.getStatus() != 200) {
+            HttpResponse response = makeRequestforTokenValidity(token);
+            StringBuilder builder = responseStringBuilder(response);
+            JSONObject jsonObject = new JSONObject(builder.toString());
+            if (response.getStatusLine().getStatusCode() != 200) {
```

Comment:
This check should be statusCode >= 200 && statusCode < 400 with follow redirects enabled for the HttpClient

Reply:
The HttpClient by default does the follow redirection.